html/user/per_app_list.php

```diff
@@ -56,74 +60,319 @@ function show_header($is_team, $apps, $appid, $is_total) {
+     global $store;
+     global $items_per_page;
+     $fetched_items = $items_per_page;
+     $check = 0;
+     $fetch = 1;
+    
+     $store[1][0] = 0;
+ 
+     //Check whether every one of the credit_teams corresponds to a valid team in the teams table
+     //and fetch accordingly bigger chunks of the credit_team table until we have the chosen size
+     //
+     if (!$up) {
+         while ($check != $fetch) {
+            $data =  BoincCreditTeam::enum("appid=$appid order by $x desc limit $used_offset,$fetched_items");
+            $check = sizeof($data) - ($fetched_items - $items_per_page);
+            $fetch = $check + ($fetched_items - $items_per_page);
+            foreach ($data as $item) {
+                     $team = BoincTeam::lookup_id($item->teamid);
+                     if (!is_object($team)) {
+                       $fetch--;
+                     }
             }
-            if (!$c) {
-                $c = new StdClass;
-                $c->expavg = 0;
-                $c->total = 0;
+            if ($check != $fetch) {
+                $fetched_items++;
             }
+         }
+         //How many teams did we get
+         //
+         $store[0][0] = $fetch;
+         
+         //How many more than the predefined items_per_page did we fetch
+         //
+         $store[1][0] = $fetched_items - $items_per_page;
+     
+     }   else {
+
+         //Do the opposite. Fetch chunks from the credit_team table but from an earlier point 
+         //every time we find a non valid team entry. 
+         //
+         while ($check != $fetch) {
+               if ($used_offset < 0) $used_offset = 0;
+               $data =  BoincCreditTeam::enum("appid=$appid order by $x desc limit $used_offset,$fetched_items");
+               $check = sizeof($data) - ($fetched_items - $items_per_page);
+               $fetch = $check + ($fetched_items - $items_per_page);
+               foreach ($data as $item) {
+                      $team = BoincTeam::lookup_id($item->teamid);
+                      if (!is_object($team)) {
+                        $fetch--;
+                      }
+             }
+             if ($check != $fetch) {
+                 $fetched_items++;
+                 $used_offset--;
+             }
+        } 
+        $store[0][0] = $fetch;
+        $store[1][0] = $fetched_items - $items_per_page;
         }
-        echo "<td align=right>".format_credit($c->expavg)."</td><td align=right>".format_credit_large($c->total)."</td>\n";
-    }
-    echo "</tr>\n";
+
+       return $data;
+
+}
+
+function choose_users($x, $appid, $used_offset, $up) {
+     global $store;
+     global $items_per_page;
+     $fetched_items = $items_per_page;
+     $check = 0;
+     $fetch = 1;
+  
+      //Do the same for the credit_user and user tables
+      //
+     if (!$up) {
+      while ($check != $fetch) {
+             $data = BoincCreditUser::enum("appid=$appid order by $x desc limit $used_offset,$fetched_items");
```

Comment:
This looks like tab indentation. BOINCs Coding style for this part of the source code is to use 4 spaces per indentation level. Please adjust your editor accordingly.